packages/neon_framework/packages/neon_storage/test/sqlite/multi_table_database_test.dart

```diff
@@ -0,0 +1,184 @@
+
+void main() {
+  group(MultiTableDatabase, () {
+    late _DatabaseController controller;
+    late MultiTableDatabase database;
+
+    setUpAll(() {
+      registerFallbackValue(_FakeBatch());
+      registerFallbackValue(_FakeMultiTableDatabase());
+    });
+
+    setUp(() {
+      controller = _DatabaseControllerMock();
+      when(() => controller.tables).thenReturn([]);
+      when(() => controller.path).thenReturn(inMemoryDatabasePath);
+      when(() => controller.name).thenReturn('database');
+
+      database = _TestDatabase(controller);
+      resetMocktailState();
+    });
+
+    tearDown(() async {
+      await database.close();
+    });
+
+    test('init initializes the database', () async {
+      await database.init();
+
+      expect(
+        database.database,
+        isA<Database>(),
+      );
+    });
+
+    test('throws if not initialized', () async {
+      expect(
+        () => database.database,
+        throwsStateError,
+      );
+    });
+
+    test('returns the same instance once initialized', () async {
+      await database.init();
+
+      expect(
+        identical(database.database, database.database),
+        isTrue,
+      );
+    });
+
+    group('conflicting table names', () {
+      setUp(() {
+        controller = _DatabaseControllerMock();
+        when(() => controller.name).thenReturn('database');
+      });
+
+      test('throws an ArgumentError for duplicate table names', () {
+        final table = _TableMock();
+        when(() => table.name).thenReturn('cache_table');
+
+        final table2 = _TableMock();
+        when(() => table2.name).thenReturn('cache_table');
+
+        when(() => controller.tables).thenReturn([table, table2]);
+
+        expect(
+          () => _TestDatabase(controller),
+          throwsArgumentError,
+        );
+      });
+
+      test('throws a TableNameError for conflicting table names', () {
+        final table = _TableMock();
+        when(() => table.name).thenReturn('_database_meta');
+
+        when(() => controller.tables).thenReturn([table]);
+
+        expect(
+          () => _TestDatabase(controller),
+          throwsArgumentError,
+        );
+      });
+    });
+
+    group('table versioning', () {
+      late Table table;
+      setUp(() async {
+        table = _TableMock();
+        when(() => table.name).thenReturn('table');
+        when(() => table.version).thenReturn(2);
+        when(() => table.onOpen()).thenAnswer((_) async {});
+
+        controller = _DatabaseControllerMock();
+        when(() => controller.path).thenReturn(inMemoryDatabasePath);
+        when(() => controller.name).thenReturn('database');
+        when(() => controller.tables).thenReturn([table]);
+
+        database = _TestDatabase(controller);
+
+        await database.init();
+        clearInteractions(table);
+      });
+
+      tearDown(() {
+        verify(table.onOpen).called(1);
+
+        verify(() => table.name);
+        verify(() => table.version);
```

Comment:
This doesn't do anything?

Reply:
This is needed for the `verifyNoMoreInteractions` that chacks that every call is covered by a verify.
We do not explicitly verify that `name` and `version` are called so we need this.